phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java

```diff
@@ -188,7 +185,17 @@ public static void waitForTaskState(Connection conn, PTable.TaskType taskType, P
 
             String taskStatus = null;
 
-            if (rs.next()) {
+            while (rs.next()) {
+                String tableName = rs.getString(PhoenixDatabaseMetaData.TABLE_NAME);
+                if (!tableName.equals(expectedTableName)) {
```

Comment:
I think it would be better to filter for the table name in the SELECT statement above (i.e. add it to the WHERE clause)

Reply:
+1 to that. I think that would also not require us to have the `continue` statements in case there are no matches.

Reply:
@gokceni can you please address these comments?

Reply:
addressed